builtin/logical/aws/client.go

```diff
@@ -44,6 +50,26 @@ func getRootConfig(ctx context.Context, s logical.Storage, clientType string, lo
 		case clientType == "sts" && config.STSEndpoint != "":
 			endpoint = *aws.String(config.STSEndpoint)
 		}
+
+		if config.IdentityTokenAudience != "" {
```

Comment:
Output of local acceptance tests in the PR description to follow soon. Adding a full-loop acceptance test in CI is out of scope for this PR and will be included in a follow-up PR